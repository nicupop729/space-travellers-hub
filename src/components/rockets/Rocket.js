import PropTypes from 'prop-types';
import RocketBtn from '../Btns/RocketBtn';
import './rocket.css';

const Rocket = ({
  image, name, description, reserved, id,
}) => (
  <li className="Rocket">
    <img className="Rocket__img" src={image} alt="rocket-img" />
    <div className="Rocket__div">
      <h1 className="Rocket__name">{name}</h1>
      <div className="Rocket__badge_desc_div">
        {reserved && <span className="Rocket__reserved_badge">Reserved</span>}
        <p className="Rocket__description">{description}</p>
      </div>
      {reserved ? (
        <RocketBtn
          className="Rocket__btn_Cancel"
          text="Cancel Reservation"
          id={id}
        />
      ) : (
        <RocketBtn
          className="Rocket__btn_Reserve"
          text="Reserve Rocket"
          id={id}
        />
      )}
    </div>
  </li>
);

Rocket.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  id: PropTypes.number.isRequired,
};

export default Rocket;
