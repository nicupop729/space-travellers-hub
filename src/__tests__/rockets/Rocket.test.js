import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Rocket from '../../components/rockets/Rocket';

describe('snapshot testing', () => {
  test('snapshot for Rocket component', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <Rocket />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
