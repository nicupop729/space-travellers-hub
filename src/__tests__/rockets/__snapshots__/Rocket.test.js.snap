// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Rocket component 1`] = `
<li
  className="Rocket"
>
  <img
    alt="rocket-img"
    className="Rocket__img"
  />
  <div
    className="Rocket__div"
  >
    <h1
      className="Rocket__name"
    />
    <div
      className="Rocket__badge_desc_div"
    >
      <p
        className="Rocket__description"
      />
    </div>
    <button
      className="Rocket__btn_Reserve"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</li>
`;
